### ISSUE TITLE
[Bug]: Getting `No such function` error when escaped defaultable parameter exists

### ISSUE THREAD

**Issue by xlight05** [ballerina-platform/ballerina-lang#39867]
### Description

$Subject.
This is required to bump ballerina central to 2201.4.0. 

### Steps to Reproduce

```
import ballerina/http;

service / on new http:Listener(9090) {

    isolated resource function get packages(string? q = (), boolean? user\-packages = false) returns string {

        return "Hello, World!";
    }
}
```

Invoke http://localhost:9090/packages?q=anjana

### Affected Version(s)

2201.4.0

### Related area

-> Runtime

**Comment by HindujaB** [comment]
A similar issue was reported at https://github.com/ballerina-platform/ballerina-lang/issues/39515.
This is fixed in the master. PR - https://github.com/ballerina-platform/ballerina-lang/pull/39518

**Comment by HindujaB** [comment]
Closing as the PR is merged with `2201.4.x` and will be available in `2201.4.2`
